core/docz-components/src/utils/hooks.tsx

```diff
@@ -0,0 +1,33 @@
+import { useEffect } from 'react'
+
+export const useScrollLock = (): void => {
+  useEffect(() => {
+    if(document !== undefined){
+      document.body.style.overflow = 'hidden'
+    }
+    return () => {
+      if(document !== undefined) {
+        document.body.style.overflow = 'unset'
+      }
+    }
+  }, []);
+}
+
+export const useEscape = (callback: () => void): void => {
+  const keydownCallback = (event: any): void => {
+    if(event.keyCode === 27){
+      callback();
+    }
+    return
+  }
+  useEffect(() => {
+    if(document !== undefined){
```

Comment:
We are specifying the condition 2 time here, maybe we could do something like:

```ts
useEffect(() => {
  if (document === undefined) {
    return () => {}
  }

  document.addEventListener('keydown', keydownCallback, true)
  return () => document.removeEventListener('keydown', keydownCallback, true)
})
```

I find it more clear but might be personal preference 👍